src/store/mutations/mutations-model.ts

```diff
@@ -17,32 +29,19 @@ export const mutateIsFilingPaying = (state: any, isFilingPaying: boolean) => {
   state.stateModel.isFilingPaying = isFilingPaying
 }
 
-export const mutateResource = (state: any, resource: ExternalResourceIF) => {
-  state.resourceModel.externalResource = resource
-}
-
-export const mutateKeycloakRoles = (state: any, keyCloakRoles: Array<string>) => {
-  state.stateModel.tombstone.keycloakRoles = keyCloakRoles
-}
-
-export const mutateAuthRoles = (state: any, authRoles: Array<string>) => {
-  state.stateModel.tombstone.authRoles = authRoles
-}
-
 export const mutateCurrentDate = (state: any, currentDate: string) => {
   state.stateModel.currentDate = currentDate
-}
-
-export const mutateCertifyStatementResource = (state: any, certifyStatementResource: CertifyStatementIF) => {
-  state.resourceModel.certifyStatementResource = certifyStatementResource
+  if (!state.stateModel.ignoreChanges) mutateHaveChanges(state, true)
 }
 
 export const mutateCertifyState = (state: any, certifyState: CertifyIF) => {
   state.stateModel.certifyState = certifyState
+  if (!state.stateModel.ignoreChanges) mutateHaveChanges(state, true)
```

Comment:
This is a special check to allow pages to load/update data without counting that as a data change.

If you add new state data, please copy this code.

Reply:
Just for clarity, what was the driving force behind this change? I don't fully grasp the complexity of it, but it seems like a bit of duplication.

Reply:
The problem is that some pages "set" data when they mount, but the purpose of this feature is to track unsaved changed and that sabotages this.
I looked at changing some components to NOT set data when they mount but that got complicated and it would be fragile.
Instead, my solution is to temporarily ignore "haveChanges" while the pages load (ie, the period between created() and the next tick).
So, this line of code (and yes, it's duplicated in this file but it's centralized) is to only keep track of changes if "ignoreChanges" isn't set.
I thought this was a decent solution to update haveChanges centrally, but I'm open to better solutions :)

Reply:
Absolutely, i think it's a great idea! Totally fine with it and as with all things, we can always adjust things later if we feel the need 👍